src/main/scala/com/google/cloud/spark/bigquery/direct/DirectBigQueryRelation.scala

```diff
@@ -139,11 +139,24 @@ private[bigquery] class DirectBigQueryRelation(
   def getNumPartitionsRequested: Int = options.parallelism
       .getOrElse(Math.max((sizeInBytes / DEFAULT_BYTES_PER_PARTITION).toInt, 1))
 
-  private def getCompiledFilter(filters: Array[Filter]): String = {
-    // If a manual filter has been specified do not push down anything.
-    options.filter.getOrElse {
-      // TODO(pclay): Figure out why there are unhandled filters after we already listed them
-      DirectBigQueryRelation.compileFilters(handledFilters(filters))
+  // VisibleForTesting
+  private[bigquery] def getCompiledFilter(filters: Array[Filter]): String = {
+    if(options.combinePushedDownFilters) {
+      // new behaviour, fixing https://github.com/GoogleCloudPlatform/spark-bigquery-connector/issues/74
+      Seq(
+        options.filter,
+        BigQueryUtil.noneIfEmpty(DirectBigQueryRelation.compileFilters(handledFilters(filters)))
+      )
+        .flatten
+        .map(f => s"($f)")
+        .mkString(" AND ")
+    } else {
+      // old behaviour, kept for backward compatibility
+      // If a manual filter has been specified do not push down anything.
+      options.filter.getOrElse {
+        // TODO(pclay): Figure out why there are unhandled filters after we already listed them
+        DirectBigQueryRelation.compileFilters(handledFilters(filters))
```

Comment:
I think boolean and filter should be independent.

Make the boolean "passThroughSparkFilters" and there are 4 valid cases:
passThrough & filter
passThrough & no filter
no passThrough & filter
no assThrough & no filter

admittedly the user can get the last behavior with noCombine and filter = '', but it makes more sense to me to make them independent

Reply:
I may be missing something, but why are they dependent? The filter is set by the 'filter' option, and the flag is set by the 'combinePushedDownFilters' option. The flag controls how the filter option is handled when creating the filter to the BQ storage API - AND'ed with the where() filters if the flag is true, or potentially overrides them if the flag is false.

Reply:
My proposed boolean option is slightly different. It controls whether to pass through Spark Filters regardless of whether the user passes a filter.

So if the user sets the boolean to False and does not provide a filter, the connector does not pass any filters through at all, whereas in your code it would pass Spark's filters through.